### PR TITLE
Rewrite `fancyhdr` tutorial

### PR DESCRIPTION
**Pull request by marcin-serwin**
This PR considerably expands the `fancyhdr` tutorial.

### CODE REVIEW

File: book/src/basics.tex [hunk 1]
```diff
@@ -674,11 +674,35 @@ \subsection{Page Styles}
 \begin{lscommand}
 \ci{thispagestyle}\verb|{|\emph{style}\verb|}|
 \end{lscommand}
-A description how to create your own
-headers and footers can be found in \companion{} and in section~\ref{sec:fancy} on page~\pageref{sec:fancy}.
-%
-% Pointer to the Fancy headings Package description !
-%
+
+You may also control the style of the displayed page numbers. To change it use
```
Comment (oetiker) [inline book/src/basics.tex:678]:
```suggestion
You can also control the style of the displayed page numbers. To change the style use
```

File: book/src/basics.tex [hunk 2]
```diff
@@ -674,11 +674,35 @@ \subsection{Page Styles}
...
+the
+\begin{lscommand}
+  \ci{pagenumbering}[style]
+\end{lscommand}
+command, where \carg{style} is one of the styles presented in
```
Comment (oetiker) [inline book/src/basics.tex:683]:
```suggestion
command, where \carg{style} is one of the styles listed in
```

File: book/src/spec.tex [hunk 3]
```diff
@@ -101,64 +101,430 @@ \section{Indexing} \label{sec:indexing}
...
-
-\begin{figure}[!htbp]
-\begin{lined}{\textwidth}
-\begin{verbatim}
-\documentclass{book}
+\subsection{Basic commands}
+
+The \pai*{fancyhdr} package provides a few simple commands that allow you to
+customise the header and footer lines of your document. It defines an
+additional page style \cargv{fancy} that can be customised to our liking. By
```
Comment (oetiker) [inline book/src/spec.tex:108]:
```suggestion
additional page style \cargv{fancy} and a set of commands to customise it to our liking. By
```

File: book/src/spec.tex [hunk 4]
```diff
@@ -101,64 +101,430 @@ \section{Indexing} \label{sec:indexing}
...
+default it only adds a line separating header from the page body.
```
Comment (oetiker) [inline book/src/spec.tex:109]:
```suggestion
default it only adds a line separating the header from the page body.
```

File: book/src/spec.tex [hunk 5]
```diff
@@ -101,64 +101,430 @@ \section{Indexing} \label{sec:indexing}
...
-\RenewDocumentCommand{\chaptermark}{m}{%
-        \markboth{#1}{}}
+
+\begin{document}
+\noindent %!hide
+This statement is false.
+\end{document}
+\end{example}
+
+The basic command of the package is
```
Comment (oetiker) [inline book/src/spec.tex:125]:
```suggestion
The primary command of the package is
```

File: book/src/spec.tex [hunk 6]
```diff
@@ -101,64 +101,430 @@ \section{Indexing} \label{sec:indexing}
...
+\begin{lscommand}
+  \ci{fancyhf}[places][field]
+\end{lscommand}
+The \carg{places} is a comma separated list of places to put the \carg{field}
+argument. There are total of 12 different places and each is identified by a
```
Comment (oetiker) [inline book/src/spec.tex:130]:
```suggestion
The \carg{places} is a comma separated list of places where the \carg{field} should be displayed. There are total of 12 different places and each is identified by a
```

File: book/src/spec.tex [hunk 7]
```diff
@@ -101,64 +101,430 @@ \section{Indexing} \label{sec:indexing}
...
+combination of three letters:
+\begin{itemize}
+  \item The first letter tells whether it is in a header (\cargv{H}) or in a
```
Comment (oetiker) [inline book/src/spec.tex:133]:
```suggestion
  \item The first specifies whether the location is in the header (\cargv{H}) or in the
```

File: book/src/spec.tex [hunk 8]
```diff
@@ -101,64 +101,430 @@ \section{Indexing} \label{sec:indexing}
...
+  footer (\cargv{F}).
+  \item The second letter tells which part of the header\slash footer should be
+  occupied. The possible positions are left (\cargv{L}), centre (\cargv{C}) and
+  right (\cargv{R}).
```
Comment (oetiker) [inline book/src/spec.tex:137]:
```suggestion
  \item The second letter specifies the location within the header\slash footer: (\cargv{L}) for left, (\cargv{C}) for center and
 (\cargv{R}) for right.
```

File: book/src/spec.tex [hunk 9]
```diff
@@ -101,64 +101,430 @@ \section{Indexing} \label{sec:indexing}
...
+  \item The third letter tells whether the field should be printed on even
+  (\cargv{E}) or odd (\cargv{O}) pages. If the document is not two sided then
+  all pages are treated as odd.
```
Comment (oetiker) [inline book/src/spec.tex:140]:
```suggestion
  \item The third letter defines whether the field should be printed on even
  (\cargv{E}) or odd (\cargv{O}) pages. If the document is not two sided then
  all pages are treated as odd.
```

File: book/src/spec.tex [hunk 10]
```diff
@@ -101,64 +101,430 @@ \section{Indexing} \label{sec:indexing}
...
+\end{itemize}
+So for example the combination \cargv{FCE} identifies the centre part of the
+footer on even pages. If any of the letters is omitted then the identifier
+points toward all positions specifiable by the omitted letter, so for example
+\cargv{HR} means right part of the header on both odd and even pages.
```
Comment (oetiker) [inline book/src/spec.tex:145]:
```suggestion
\cargv{HR} is right side of the header on both odd and even pages.
```

File: book/src/spec.tex [hunk 11]
```diff
@@ -101,64 +101,430 @@ \section{Indexing} \label{sec:indexing}
...
+\begin{document}
+\noindent %!hide
+The next statement is false.
+The previous statement is true.
+\end{document}
+\end{example}
+
+There are two additional commands \ci{fancyhead} and \ci{fancyfoot} that work
+the same way except that they imply respectively \cargv{H} and \cargv{F} in
+their \carg{places} argument if it's missing.
```
Comment (oetiker) [inline book/src/spec.tex:176]:
```suggestion
their \carg{places} argument unless specified differently.
```

File: book/src/spec.tex [hunk 12]
```diff
@@ -101,64 +101,430 @@ \section{Indexing} \label{sec:indexing}
...
+
+\begin{document}
+\noindent %!hide
+The next statement is false.
+The previous statement is true.
+\end{document}
+\end{example}
+
+There are two additional commands \ci{fancyhead} and \ci{fancyfoot} that work
+the same way except that they imply respectively \cargv{H} and \cargv{F} in
```
Comment (oetiker) [inline book/src/spec.tex:175]:
```suggestion
the same way except that they assume \cargv{H} and \cargv{F} respectively
```

File: book/src/spec.tex [hunk 13]
```diff
@@ -101,64 +101,430 @@ \section{Indexing} \label{sec:indexing}
...
+\end{document}%!hide
+\end{example}
+
+The lines drawn by the \pai{fancyhdr} package may also be customised. To change
+their thickness redefine the
+\begin{lscommand}
+  \ci{headrulewidth} \\
+  \ci{footrulewidth}
+\end{lscommand}
+macros to the desired length.
```
Comment (oetiker) [inline book/src/spec.tex:202]:
```suggestion
macros to the desired width.
```

File: book/src/spec.tex [hunk 14]
```diff
@@ -101,64 +101,430 @@ \section{Indexing} \label{sec:indexing}
...
+all statements that are not
+dedicated to themselves. 
+\end{document}%!hide
+\end{example}
+
+It is often useful to have the header and footer contain information based on
+the contents of the page. These are called \enquote{marks} in \LaTeX{}
+terminology. Before we talk about the default ones let's consider how you can
+define your own using the \pai{extramarks}:\footnote{\pai{extramarks} is part
+of \pai*{fancyhdr}.}
```
Comment (oetiker) [inline book/src/spec.tex:264]:
```suggestion
terminology. Before we talk about the default ones let's consider how you can
define your own using the \pai{extramarks}\footnote{\pai{extramarks} is part
of \pai*{fancyhdr}.} package.
```

File: book/src/spec.tex [hunk 15]
```diff
@@ -101,64 +101,430 @@ \section{Indexing} \label{sec:indexing}
...
+\begin{lscommand}
+  \ci*{extramarks}[left:m ! right:m] \\
+  \ci{firstleftxmark} \\
+  \ci{firstrightxmark} \\
+  \ci{lastleftxmark} \\
+  \ci{lastrightxmark}
+\end{lscommand}
+The \ci{extramarks} is used to set the contents of \carg{left} and \carg{right}
```
Comment (oetiker) [inline book/src/spec.tex:272]:
```suggestion
The \ci{extramarks} command is used to set the contents of \carg{left} and \carg{right}
```

File: book/src/spec.tex [hunk 16]
```diff
@@ -101,64 +101,430 @@ \section{Indexing} \label{sec:indexing}
...
+marks. Then you can access those inside the headers by using the appropriate
```
Comment (oetiker) [inline book/src/spec.tex:273]:
```suggestion
marks. Then you can access these marks inside the headers by using the appropriate
```

File: book/src/spec.tex [hunk 17]
```diff
@@ -101,64 +101,430 @@ \section{Indexing} \label{sec:indexing}
...
+\section{Second}
+\subsection{Sub}
+\end{document}%!hide
+\end{example}
+Note that \ci{firstrightmark} is empty in the above example. This is caused by
+the fact that \ci{section} commands sets both marks, leaving the right one
+empty.
+
+You may have noticed that section titles are typeset in uppercase when given by
+marks. To disable this use the \ci{nouppercase} command inside the \ci{fancyhf}
```
Comment (oetiker) [inline book/src/spec.tex:335]:
```suggestion
You may have noticed that section titles are typeset in uppercase when provided by
marks commands. To disable this use the \ci{nouppercase} command inside the \ci{fancyhf}
```

File: book/src/spec.tex [hunk 18]
```diff
@@ -101,64 +101,430 @@ \section{Indexing} \label{sec:indexing}
...
+
+\begin{document}%!hide
+\section{Person}
+There exists a person such that
+if they are reading this then
+everybody is reading this.
+\end{document}%!hide
+\end{example}
+In case you want to change only the right mark, the \ci{markright} command is
+available.
```
Comment (oetiker) [inline book/src/spec.tex:376]:
```suggestion
If you only want to change the right mark, use the \ci{markright} command.
```

File: book/src/spec.tex [hunk 19]
```diff
@@ -101,64 +101,430 @@ \section{Indexing} \label{sec:indexing}
...
+
+By default the field in the centre of header\slash footer will expand equally
```
Comment (oetiker) [inline book/src/spec.tex:378]:
```suggestion
By default the field in the centre of the header\slash footer will expand equally
```

**Review by oetiker — CHANGES_REQUESTED** [review]

**Comment by oetiker** [comment]
great! don't forget the Changes file!

**Review by oetiker — APPROVED** [review]

**Comment by oetiker** [comment]
thank you! any more plans ?

**Comment by marcin-serwin** [comment]
I'm currently working on integrating `chktex` and `latexindent` into github actions workflow. After that I planned to rework the `realworld` and `intro` to better introduce what logical markup is. After that maybe update `fontspec` and change the layout a bit so that a4, a5 and letter versions use the same text width (fewer problems with `overfull hbox`es).

**Comment by oetiker** [comment]
Great :) looking forward to your PRs. What is motivating you to all this effort?

**Comment by marcin-serwin** [comment]
> What is motivating you to all this effort?

I don't know, it's just fun :>

**Pull request merged**